java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -104,9 +104,30 @@ private JavaType substituteTypeParameter(JavaType type, TypeSubstitution substit
         return parametrizedTypeCache.getWildcardType(substitutedType, wildCardType.boundType);
       }
     }
+    if (type instanceof JavaType.ArrayJavaType) {
+      return substituteTypeParameterInArray((JavaType.ArrayJavaType) type, substitution);
+    }
     return type;
   }
 
+  private JavaType substituteTypeParameterInArray(JavaType.ArrayJavaType arrayType, TypeSubstitution substitution) {
+    JavaType rootElementType = arrayType.elementType;
+    int nbDimensions = 1;
+    while (rootElementType.isTagged(JavaType.ARRAY)) {
+      rootElementType = ((JavaType.ArrayJavaType) rootElementType).elementType;
+      nbDimensions++;
+    }
+    JavaType substitutedType = substitution.substitutedType(rootElementType);
+    if (substitutedType != null) {
+      // FIXME SONARJAVA-1574 a new array type should not be created but reused if already existing for the current element type
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1134)